### PR TITLE
chore: stubs instruction

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: scripts/tools/install_stubs.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+LIBRARIES=(cloudpickle deepmerge fs huggingface_hub importlib_metadata joblib multipart pandas pathspec PIL starlette transformers xgboost yaml)
```
Comment (parano) [inline scripts/tools/install_stubs.sh:3]: should we have a requirements.txt file just for typing, that locks to a version of these libraries?

File: .gitignore [hunk 2]
```diff
@@ -109,3 +109,5 @@ tests/integration/data
 
 # setuptools_scm generated version file
 bentoml/_version.py
+typings/pandas
+typings/numpy
```
Comment (parano) [inline .gitignore:113]: we will need to add all the other libs in the list below?

maybe let's ignore everything under `typings` here?  and when we actually need to commit some stub files to the repo, we can manually do that.

Comment (aarnphm) [reply .gitignore:113]: sure

File: scripts/tools/install_stubs.sh (hunk 1 above)
Comment (aarnphm) [reply scripts/tools/install_stubs.sh:3]: sure we can do that

Comment (aarnphm) [reply scripts/tools/install_stubs.sh]: done

File: .gitignore (hunk 2 above)
Comment (aarnphm) [reply .gitignore]: done

**Comment by sauyon** [comment]
What was the point of generating stubs again?

**Comment by aarnphm** [comment]
> What was the point of generating stubs again?

For local use with pyright.

**Comment by sauyon** [comment]
Does pyright not pick up the types from the libraries? I was under the impression type stubs were only required if the typing was incomplete or non-existent.

**Comment by aarnphm** [comment]
> Does pyright not pick up the types from the libraries? I was under the impression type stubs were only required if the typing was incomplete or non-existent.

not for most ml frameworks

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -302,6 +279,17 @@ BentoML also provides git hooks that developers can install with:
 ```bash
 make hooks
 ```
+## Stubs
+Refers to [Installation](https://github.com/microsoft/pyright#installation) to install pyright correctly.
+
+In order to make pyright function correctly one also need to run the following scripts alongside with the stubs provided
+ in the main repository.
+```bash
+./script/tools/install_stubs.sh
```
Comment (bojiang) [inline DEVELOPMENT.md:288]: where is the install_stubs.sh ?

File: scripts/ci/helpers.sh [hunk 4]
```diff
@@ -9,6 +9,10 @@ PASS() {
     echo -e "$GREEN""[PASS]""$NC" "$*"
 }
 
+INFO() {
+    echo -e "$YELLOW""[INFO]""$NC" "$*"
```
Comment (sauyon) [inline scripts/ci/helpers.sh:13]: Personally find this to look nicer.

```suggestion
    echo -e "${YELLOW}[INFO]${NC}" "$*"
```

Or alternatively just

```suggestion
    echo -e "$YELLOW[INFO]$NC" "$*"
```

**Comment by sauyon** [comment]
lol github didn't show the review comments I had on the nonexistent file

**Pull request merged**